File: tests/domains/video/test_cut.py
import pytest

from domains import Runtime
from domains.video.cut import cut_video
from domains.video.probe import probe_video
from tests.helpers import unwrap, unwrap_job

pytestmark = pytest.mark.ffmpeg


def test_cut_creates_file(runtime: Runtime, sample_video: str) -> None:
    result = unwrap(cut_video(runtime, sample_video, 0.5, 2.0, reencode=True))
    assert result["output"] == "sample_cut_0.5-2.mp4"
    probed = unwrap(probe_video(runtime, result["output"]))
    assert abs(probed["info"]["duration"] - 1.5) < 0.2


def test_cut_invalid_range(runtime: Runtime, sample_video: str) -> None:
    result = cut_video(runtime, sample_video, 2.0, 1.0)
    assert result.get("code") == "invalid_argument"


def test_cut_beyond_duration(runtime: Runtime, sample_video: str) -> None:
    result = cut_video(runtime, sample_video, 0.0, 99.0)
    assert result.get("code") == "invalid_argument"


def test_cut_missing_file(runtime: Runtime) -> None:
    result = cut_video(runtime, "ghost.mp4", 0.0, 1.0)
    assert result.get("code") == "not_found"


def test_cut_background(runtime: Runtime, sample_video: str) -> None:
    submitted = unwrap_job(cut_video(runtime, sample_video, 0.0, 1.0, background=True))
    job = runtime.jobs.wait(submitted["job_id"], timeout=30)
    assert job.status == "done"
    assert job.result is not None
    assert str(job.result["output"]).startswith("sample_cut_")

File: tests/domains/video/test_cut.rs
use std::time::Duration;

use mcp_tools::core::errors::ErrorCode;
use mcp_tools::core::jobs::JobStatus;
use mcp_tools::domains::video::cut::cut_video;
use mcp_tools::domains::video::probe::probe_video;

use crate::conftest::{runtime, sample_video};
use crate::helpers::{unwrap, unwrap_job};
use crate::skip_without_ffmpeg;

#[test]
fn test_cut_creates_file() {
    skip_without_ffmpeg!();
    let rt = runtime();
    let sample = sample_video(&rt.runtime.workspace);
    let result = unwrap(cut_video(&rt.runtime, &sample, 0.5, 2.0, true, false));
    assert_eq!(result.output, "sample_cut_0.5-2.mp4");
    let probed = probe_video(&rt.runtime, &result.output).unwrap();
    assert!((probed.info.duration - 1.5).abs() < 0.2);
}

#[test]
fn test_cut_invalid_range() {
    skip_without_ffmpeg!();
    let rt = runtime();
    let sample = sample_video(&rt.runtime.workspace);
    let error = cut_video(&rt.runtime, &sample, 2.0, 1.0, false, false).unwrap_err();
    assert_eq!(error.code, ErrorCode::InvalidArgument);
}

#[test]
fn test_cut_beyond_duration() {
    skip_without_ffmpeg!();
    let rt = runtime();
    let sample = sample_video(&rt.runtime.workspace);
    let error = cut_video(&rt.runtime, &sample, 0.0, 99.0, false, false).unwrap_err();
    assert_eq!(error.code, ErrorCode::InvalidArgument);
}

#[test]
fn test_cut_missing_file() {
    let rt = runtime();
    let error = cut_video(&rt.runtime, "ghost.mp4", 0.0, 1.0, false, false).unwrap_err();
    assert_eq!(error.code, ErrorCode::NotFound);
}

#[test]
fn test_cut_background() {
    skip_without_ffmpeg!();
    let rt = runtime();
    let sample = sample_video(&rt.runtime.workspace);
    let submitted = unwrap_job(cut_video(&rt.runtime, &sample, 0.0, 1.0, false, true));
    let job = rt
        .runtime
        .jobs
        .wait(&submitted.job_id, Some(Duration::from_secs(30)))
        .unwrap();
    assert_eq!(job.status, JobStatus::Done);
    let result = job.result.expect("resultado");
    assert!(result["output"]
        .as_str()
        .unwrap()
        .starts_with("sample_cut_"));
}
